src/resources/2014/ConspiracyIsland.ts

```diff
@@ -0,0 +1,296 @@
+    requiredItem: Item,
+    maximise: string
+  ) {
+    this.name = name;
+    this.reward = reward;
+    this.questStateProperty = questStateProperty;
+    this.questProgressLimit = questProgressLimit;
+    this.questProgressProperty = questProgressProperty;
+    this.canUseWanderers = canUseWanderers;
+    this.canUseRunaways = canUseRunaways;
+    this.cleeshsafe = cleeshSafe;
+    this.questLocation = questLocation;
+    this.requiredItem = requiredItem;
+    this.maximise = maximise;
+  }
+
+  currentQuest(): boolean {
+    return get(this.questStateProperty) !== "unstarted";
+  }
+}
+
+export const quests = [
+  new QuestData(
+    "Serum",
+    30,
+    "questESpSerum",
+    5,
+    "",
+    false,
+    false,
+    false,
+    $location`The Mansion of Dr. Weirdeaux`,
+    $item`none`,
+    "10 item, 10 pickpocket chance"
+  ),
+  new QuestData(
+    "Clipper",
+    20,
+    "questESpClipper",
+    23,
+    "fingernailsClipped",
+    false,
+    true,
+    false,
+    $location`The Mansion of Dr. Weirdeaux`,
+    $item`none`,
+    ""
+  ),
+  new QuestData(
+    "EVE",
+    30,
+    "questESpEVE",
+    0,
+    "",
+    true,
+    true,
+    true,
+    $location`The Secret Government Laboratory`,
+    $item`none`,
+    ""
+  ),
+  new QuestData(
+    "FakeMedium",
+    30,
+    "questESpFakeMedium",
+    0,
+    "",
+    true,
+    true,
+    true,
+    $location`The Secret Government Laboratory`,
+    $item`none`,
+    ""
+  ),
+  new QuestData(
+    "Gore",
+    20,
+    "questESpGore",
+    100,
+    "goreCollected",
+    true,
+    false,
+    false,
+    $location`The Secret Government Laboratory`,
+    $item`gore bucket`,
+    "10 meat"
+  ),
+  new QuestData(
+    "JunglePun",
+    20,
+    "questESpJunglePun",
+    11,
+    "junglePuns",
+    true,
+    false,
+    false,
+    $location`The Deep Dark Jungle`,
+    $item`encrypted micro-cassette recorder`,
+    "mys"
+  ),
+  new QuestData(
+    "OutOfOrder",
+    30,
+    "questESpOutOfOrder",
+    99,
+    "",
+    true,
+    false,
+    false,
+    $location`The Deep Dark Jungle`,
+    $item`GPS-tracking wristwatch`,
+    "10 Initiative"
+  ),
+  new QuestData(
+    "Smokes",
+    30,
+    "questESpSmokes",
+    10,
+    "",
+    false,
+    false,
+    false,
+    $location`The Deep Dark Jungle`,
+    $item`none`,
+    "10 item, 10 pickpocket chance"
+  ),
+] as const;
+
+export function available(): boolean {
+  return have() || get("_spookyAirportToday");
+}
+
+const BLANK_QUEST = new QuestData(
+  "",
+  0,
+  "",
+  0,
+  "",
+  false,
+  false,
+  false,
+  $location`none`,
+  $item`none`,
+  ""
+);
+
+export function completedOneTimeQuests(): boolean {
+  return quests
+    .filter((q) => q.reward === 30)
+    .every((q) => get(q.questStateProperty) === "finished");
+}
+
+export function hasQuest(): boolean {
+  return quests.some((q) => q.currentQuest());
+}
+
+export function activeQuest(): QuestData {
+  return quests.find((q) => q.currentQuest()) || BLANK_QUEST;
+}
```

Comment:
Rather than having these be two separate functions, it may be nicer to have this be only the `activeQuest` function and have it return `Quest | undefined`. undefined is falsy, so you get the same information out of it.

Reply:
Ambivalent on this